### PR TITLE
JitArm64: Merge ps_mulsX, ps_maddXX, and parts of fp_arith

### PR DESCRIPTION
**Pull request by JosJuice**
Some refactorization to get rid of duplicated code. Preparation for implementing AccurateNaNs in JitArm64.

### CODE REVIEW

**Comment by Rumi-Larry** [comment]
I read in the progress report that the ARM JIT had already reached feature parity.

**Comment by JosJuice** [comment]
The progress report did say that, but with the following qualifier: "This means that **every PowerPC instruction** that the x86-64 JIT supports along with **every major JIT feature** are now supported in the AArch64 JIT!" That does not mean **every** feature. Jit64 has some minor features that JitArm64 lacks, and vice versa too.

To be more specific regarding AccurateNaNs: There's only one game that needs that setting, and for a good while, that game was working properly in JitArm64 without doing anything special, simply because AArch64's default NaN is the same as PowerPC's default NaN (unlike x64). However, when I made some accuracy improvements to how JitArm64 handles FMA rounding, I added a host instruction that flips the sign bit of the output of certain PowerPC FMA instructions, and that host instruction flips the sign bit of NaNs as well, meaning that for the affected PowerPC instructions, the default NaN effectively has the wrong value for the sign bit from the perspective of the guest software. I was planning on explaining this in the follow-up PR that actually adds support for AccurateNaNs to JitArm64 rather than in this refactorization PR.

File: Source/Core/Core/PowerPC/JitArm64/JitArm64_FloatingPoint.cpp [hunk 1]
```diff
@@ -75,173 +75,120 @@ void JitArm64::fp_arith(UGeckoInstruction inst)
...
-
-    if (round_c)
-    {
-      ASSERT_MSG(DYNA_REC, !inputs_are_singles, "Tried to apply 25-bit precision to single");
+    ASSERT_MSG(DYNA_REC, !inputs_are_singles, "Tried to apply 25-bit precision to single");
 
+    V0Q = fpr.GetReg();
+    V1Q = fpr.GetReg();
+
+    Force25BitPrecision(reg_encoder(V1Q), VC, reg_encoder(V0Q));
```
Comment (degasus) [inline Source/Core/Core/PowerPC/JitArm64/JitArm64_FloatingPoint.cpp:114]: needs a rebase

**Review by degasus — APPROVED** [review]
looks good. However hard to track if nothing was missed, so better to merge after the next beta release.

**Pull request merged**